src/sentry/api/serializers/rest_framework/sentry_app.py

```diff
@@ -34,3 +34,16 @@ class SentryAppSerializer(Serializer):
     redirectUrl = serializers.URLField(required=False)
     isAlertable = serializers.BooleanField(required=False)
     overview = serializers.CharField(required=False)
+
+    def validate_events(self, attrs, source):
+        if not attrs.get('scopes'):
+            return attrs
+
+        for resource in attrs[source]:
```

Comment:
Could you hit a keyerror here if someone requests invalid sources?

Reply:
@markstory 

> If empty values imply no update how do you discern an omitted key and intentionally blanking the scopes?

So in creating an app, scopes are required and the `ApiScopesField` checks for that. In the case where you are updating, the mediator checks the updated events with the current scope permissions of the app https://github.com/getsentry/sentry/pull/11576/files#diff-36ac307721448b98d7d06dd7c0fdcc6fR49

(hopefully I understood your question and this answers it, if not lemme kno)

Reply:
also whoops should have replied in the other thread sorry